### PR TITLE
Add custom values file for scalardb-graphql Helm chart

### PR DESCRIPTION
**Pull request by ymorimo**
This PR adds a custom values file for scalardb-graphql Helm chart.

This file is going to be referenced from the deployment guide. scalar-labs/scalardb-graphql#18

Currently, it covers AWS only. The database configurations and the Ingress annotations for other services should be added when needed.

### CODE REVIEW

File: conf/scalardb-graphql-custom-values.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    # ALB
+    alb.ingress.kubernetes.io/scheme: internal
+    alb.ingress.kubernetes.io/target-group-attributes: stickiness.enabled=true,stickiness.lb_cookie.duration_seconds=60
+    alb.ingress.kubernetes.io/target-type: ip
+    alb.ingress.kubernetes.io/healthcheck-path: /graphql?query=%7B__typename%7D
+  hosts:
+    - host: ""
+      paths:
+        - path: /graphql
+          pathType: Exact
```
Comment (ymorimo) [inline conf/scalardb-graphql-custom-values.yaml:30]: `pathType: Exact` is enough for the GraphQL single endpoint. And I confirmed this works with ALB.
https://kubernetes.io/docs/concepts/services-networking/ingress/#path-types

Screen Shot 2022-07-01 at 20 10 48

**Review by kota2and3kan — COMMENTED** [review]
Overall looks good to me, but I have comments about `tolerations` and `podAntiAffinity`.

---

### 1. About `tolerations`

It seems that there is no configuration of `tolerations` like `scalardl-custom-values.yaml` and `scalardb-custom-values.yaml`.

https://github.com/scalar-labs/scalar-kubernetes/blob/master/conf/scalardl-custom-values.yaml#L99-L103
https://github.com/scalar-labs/scalar-kubernetes/blob/master/conf/scalardb-custom-values.yaml#L86-L90

I think some non Scalar DB GraphQL Server pods (e.g. user application pod) will be deployed to the node group for Scalar DB GraphQL Server, if there is no configuration of `tolerations`.
So, I think it is better to add `tolerations` to this sample custom values file.
(Also, in this case, we need to add the section that explain about adding `Taints` to the Kubernetes worker node in the deployment guide.)
What do you think?

---

### 2. About `podAntiAffinity`

It seems that there is no configuration of `podAntiAffinity` like `scalardl-custom-values.yaml` and `scalardb-custom-values.yaml`.
https://github.com/scalar-labs/scalar-kubernetes/blob/master/conf/scalardb-custom-values.yaml#L101-L115
https://github.com/scalar-labs/scalar-kubernetes/blob/master/conf/scalardl-custom-values.yaml#L114-L128

I think several Scalar DB GraphQL pods will be deployed to the same one node, if there is no configuration of `podAntiAffinity`.
So, I think it is better to add `podAntiAffinity` to distribute pods within the Kubernetes cluster.
What do you think?

**Comment by kota2and3kan** [comment]
@ymorimo 
> I have added them in ff82496 and I'm going to add a description about `Taints` to the deployment guide right away.

Thank you for the update!
For your reference, I created the PR about `Taints` in https://github.com/scalar-labs/scalar-kubernetes/pull/183.

**Comment by ymorimo** [comment]
@kota2and3kan Sorry for the late reply. I have added a description about Taints in scalar-labs/scalardb-graphql#32. Please take a look.

**Review by kota2and3kan — APPROVED** [review]
LGTM! Thank you!

---

It worked well in my local environment!

* tolerations
  ```console
  $ kubectl describe node minikube | grep Taints
  Taints:             <none>
  
  $ kubectl describe node minikube-m02 | grep Taints
  Taints:             kubernetes.io/app=scalardbgraphqlpool:NoSchedule
  ```
  ```console
  $ kubectl get pod -o wide
  NAME                                READY   STATUS    RESTARTS   AGE   IP           NODE           NOMINATED NODE   READINESS GATES
  mysql-scalardb-0                    1/1     Running   0          50m   172.17.0.4   minikube       <none>           <none>
  postgresql-scalardb-0               1/1     Running   0          50m   172.17.0.5   minikube       <none>           <none>
  scalardb-6644875756-28fld           1/1     Running   0          48m   172.17.0.7   minikube       <none>           <none>
  scalardb-envoy-84bf44bc5d-84c88     1/1     Running   0          48m   172.17.0.6   minikube       <none>           <none>
  scalardb-graphql-7db6cdd8b5-8596x   0/1     Running   0          2s    172.17.0.4   minikube-m02   <none>           <none>
  scalardb-graphql-7db6cdd8b5-vg9k5   0/1     Running   0          3s    172.17.0.3   minikube-m02   <none>           <none>
  scalardb-graphql-7db6cdd8b5-zls4b   0/1     Running   0          2s    172.17.0.2   minikube-m02   <none>           <none>
  ```
  * Scalar DB GraphQL pods run on `minikube-m02` and other pods run on `minikube`. In other words, I was able to make `minikube-m02` dedicated node for Scalar DB GraphQL.

* podAntiAffinity (I use `requiredDuringSchedulingIgnoredDuringExecution` for testing.)
  ```console
  $ kubectl get pod -o wide
  NAME                                READY   STATUS    RESTARTS   AGE   IP           NODE           NOMINATED NODE   READINESS GATES
  mysql-scalardb-0                    1/1     Running   0          52m   172.17.0.4   minikube       <none>           <none>
  postgresql-scalardb-0               1/1     Running   0          51m   172.17.0.5   minikube       <none>           <none>
  scalardb-6644875756-28fld           1/1     Running   0          49m   172.17.0.7   minikube       <none>           <none>
  scalardb-envoy-84bf44bc5d-84c88     1/1     Running   0          49m   172.17.0.6   minikube       <none>           <none>
  scalardb-graphql-7bf5c4744c-52gc4   0/1     Pending   0          2s    <none>       <none>         <none>           <none>
  scalardb-graphql-7bf5c4744c-qrcpv   0/1     Running   0          2s    172.17.0.2   minikube-m02   <none>           <none>
  scalardb-graphql-7bf5c4744c-xfmtt   0/1     Pending   0          2s    <none>       <none>         <none>           <none>
  ```
  * Only one Scalar DB GraphQL pod runs on one node.

File: conf/scalardb-graphql-custom-values.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
+replicaCount: 3
+
+scalarDbGraphQlConfiguration:
+  #
+  # DynamoDB
```
Comment (feeblefakie) [inline conf/scalardb-graphql-custom-values.yaml:5]: I think it looks like it's usable only for DynamoDB. (understand you took it from the existing custom confg.)

```suggestion
  # Specify the configuration of a database you use.
  # The following is the example configuration of DynamoDB.
```

**Review by feeblefakie — APPROVED** [review]
Left a minor suggestion.
Other than that, LGTM! Thank you!

**Review by brfrn169 — COMMENTED** [review]
LGTM! Thank you!

**Review by brfrn169 — APPROVED** [review]

File: conf/scalardb-graphql-custom-values.yaml (hunk 2 above)
Comment (brfrn169) [reply conf/scalardb-graphql-custom-values.yaml:5]: Ping @ymorimo

Comment (ymorimo) [reply conf/scalardb-graphql-custom-values.yaml]: @brfrn169 Thank you for notifying me!

**Pull request merged**